### DISCUSSION TITLE
What is the difference between Code Server and VS Code Web

### DISCUSSION THREAD

**Discussion by cuining** [coder/code-server#3523] [Q&A] [answered]
Hi,

I'm a beginner and want to know the difference between the  two, here is a [link](https://code.visualstudio.com/updates/v1_40#_test-vs-code-running-in-a-browser) 

Thanks in advance!

**Comment by jsjoeio** [comment, marked as answer]
Based on the link you gave, it says for VS Code Web:

> Note: Do not use this setup for any serious development. The intent is to allow the community to contribute changes back to VS Code to improve the overall experience in the browser.

Whereas code-server _is_ for serious development.  It's similar to other web IDEs like StackBlitz, Codesandbox, Replit.

Really, the main thing between VS Code and code-server is being able to access your editor from a web browser. This means:
- you could host your code-server instance on a remote machine and access from anywhere
- you could access via an Android phone/tablet or an iPad

The possibilities are endless! Hope that helps :)

**Reply by cuining** [reply]
understood. Thank U very much.

I tried it yesterday and found that the extensions cannot be installed. Is this a bug ?

**Reply by jsjoeio** [reply]
Hmm...something funky could be happening. Installing extensions works for me. Can you open a [bug report](https://github.com/cdr/code-server/issues/new?assignees=&labels=&template=bug-report.md&title=)?

**Reply by cuining** [reply]
It may be caused by the network...

**Reply by jsjoeio** [reply]
Could be! We do have some [docs](https://github.com/cdr/code-server/blob/main/docs/FAQ.md#installing-an-extension) around installing extensions manually and changing the marketplace as well. I hope those help!
